test/extended/dr/machine_recover.go

```diff
@@ -102,12 +102,12 @@ var _ = g.Describe("[sig-cluster-lifecycle][Feature:DisasterRecovery][Disruptive
 				// o.Expect(err).NotTo(o.HaveOccurred())
 
 				targets := []*corev1.Node{ /* replacedMaster , */ replacedWorker}
-				targetMachineNames := []string{ /* replacedMasterMachineName, */ replacedWorkerMachineName}
+				targetMachineNames := sets.NewString( /* replacedMasterMachineName, */ replacedWorkerMachineName)
 
 				// we use a hard shutdown to simulate a poweroff
 				for _, target := range targets {
 					framework.Logf("Forcing shutdown of node %s", target.Name)
-					_, err = ssh("sudo -i systemctl poweroff --force --force", target)
+					execOnNodeOrFail(target, "sudo -i systemctl poweroff --force --force")
```

Comment:
Retrying this operation (as seen from the logs) is not correct - because you are shutting down the node it's going to keep retrying, and it shouldn't have to:

```
Dec  8 06:18:56.307: INFO: Forcing shutdown of node ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:18:56.307: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:18:56.307: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
Dec  8 06:19:15.497: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:19:15.497: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:19:15.498: INFO: ssh core@10.0.155.158:22: stderr:    "Powering off.\n"
Dec  8 06:19:15.498: INFO: ssh core@10.0.155.158:22: exit code: 0
Dec  8 06:19:20.498: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:19:20.498: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
error dialing core@ae845ee70baa5457cb71e56356b4f460-523924702.us-west-1.elb.amazonaws.com:22: 'ssh: handshake failed: EOF', retrying
Dec  8 06:19:25.894: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:19:25.894: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:19:25.894: INFO: ssh core@10.0.155.158:22: stderr:    ""
Dec  8 06:19:25.894: INFO: ssh core@10.0.155.158:22: exit code: 0
Dec  8 06:19:30.498: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:19:30.498: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
error dialing core@ae845ee70baa5457cb71e56356b4f460-523924702.us-west-1.elb.amazonaws.com:22: 'ssh: handshake failed: EOF', retrying
Dec  8 06:19:35.813: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:19:35.813: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:19:35.813: INFO: ssh core@10.0.155.158:22: stderr:    ""
Dec  8 06:19:35.813: INFO: ssh core@10.0.155.158:22: exit code: 0
Dec  8 06:19:40.498: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:19:40.498: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
error dialing core@ae845ee70baa5457cb71e56356b4f460-523924702.us-west-1.elb.amazonaws.com:22: 'ssh: handshake failed: EOF', retrying
Dec  8 06:19:45.940: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:19:45.940: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:19:45.940: INFO: ssh core@10.0.155.158:22: stderr:    ""
Dec  8 06:19:45.940: INFO: ssh core@10.0.155.158:22: exit code: 0
Dec  8 06:19:50.498: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:19:50.498: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
error dialing core@ae845ee70baa5457cb71e56356b4f460-523924702.us-west-1.elb.amazonaws.com:22: 'ssh: handshake failed: EOF', retrying
Dec  8 06:19:55.922: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:19:55.922: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:19:55.922: INFO: ssh core@10.0.155.158:22: stderr:    ""
Dec  8 06:19:55.922: INFO: ssh core@10.0.155.158:22: exit code: 0
Dec  8 06:20:00.498: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:20:00.498: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
error dialing core@ae845ee70baa5457cb71e56356b4f460-523924702.us-west-1.elb.amazonaws.com:22: 'ssh: handshake failed: EOF', retrying
Dec  8 06:20:05.852: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:20:05.852: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:20:05.852: INFO: ssh core@10.0.155.158:22: stderr:    ""
Dec  8 06:20:05.852: INFO: ssh core@10.0.155.158:22: exit code: 0
Dec  8 06:20:10.498: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:20:10.498: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
error dialing core@ae845ee70baa5457cb71e56356b4f460-523924702.us-west-1.elb.amazonaws.com:22: 'ssh: handshake failed: EOF', retrying
Dec  8 06:20:15.899: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:20:15.899: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:20:15.899: INFO: ssh core@10.0.155.158:22: stderr:    ""
Dec  8 06:20:15.899: INFO: ssh core@10.0.155.158:22: exit code: 0
Dec  8 06:20:15.899: INFO: Getting external IP address for ip-10-0-155-158.us-west-1.compute.internal
Dec  8 06:20:15.899: INFO: SSH "sudo -i systemctl poweroff --force --force" on ip-10-0-155-158.us-west-1.compute.internal(10.0.155.158:22)
error dialing core@ae845ee70baa5457cb71e56356b4f460-523924702.us-west-1.elb.amazonaws.com:22: 'ssh: handshake failed: EOF', retrying
Dec  8 06:20:21.216: INFO: ssh core@10.0.155.158:22: command:   sudo -i systemctl poweroff --force --force
Dec  8 06:20:21.216: INFO: ssh core@10.0.155.158:22: stdout:    ""
Dec  8 06:20:21.216: INFO: ssh core@10.0.155.158:22: stderr:    ""
Dec  8 06:20:21.216: INFO: ssh core@10.0.155.158:22: exit code: 0
STEP: Disruption complete; stopping async validations
```

Reply:
Yeah, I noticed that. It's not from this test, though, as per the leading comment it's from https://github.com/openshift/origin/pull/25707.

Reply:
As per a [comment on your PR](https://github.com/openshift/origin/pull/25750#discussion_r538968242), I've added special handling for shutdown initiation in #25707 to account for the shutdown command always terminating with an error on success due to the connection being closed by the server.